### PR TITLE
Remove public fields for pauli ops

### PR DESCRIPTION
**Pull request by daxfohl**
https://github.com/quantumlib/Cirq/issues/4851 for all the pauli stuff.

I skipped BaseDensePauliString because including `cirq._compat` there makes a weird circular dependency.

### CODE REVIEW

**Comment by MichaelBroughton** [comment]
Did the `LazyLoader` not work in this case ?

**Comment by daxfohl** [comment]
@MichaelBroughton Didn't try, but fixed it in #5064 by moving the dependency, which I think puts it in a better location anyway. If you have reservations about that, then let's continue that conversation on that issue.

File: cirq-core/cirq/ops/pauli_measurement_gate.py [hunk 1]
```diff
@@ -68,17 +69,36 @@ def __init__(
...
+    @key.setter  # type: ignore
+    @deprecated(
+        deadline="v0.15",
+        fix="The mutators of this class are deprecated, instantiate a new object instead.",
+    )
+    def key(self, key: Union[str, 'cirq.MeasurementKey']):
+        if isinstance(key, value.MeasurementKey):
+            self._mkey = key
+        else:
+            self._mkey = value.MeasurementKey(name=key)
```
Comment (MichaelBroughton) [inline cirq-core/cirq/ops/pauli_measurement_gate.py:89]: previous logic seemed a little tighter here. Why change ?

**Review by MichaelBroughton — APPROVED** [review]
No worries. LGTM after nit.

File: cirq-core/cirq/ops/pauli_measurement_gate.py (hunk 1 above)
Comment (daxfohl) [reply cirq-core/cirq/ops/pauli_measurement_gate.py]: done

**Pull request merged**